src/qibo/core/circuit.py

```diff
@@ -128,7 +128,7 @@ def compile(self):
             raise_error(RuntimeError, "Circuit is already compiled.")
         if not self.queue:
             raise_error(RuntimeError, "Cannot compile circuit without gates.")
-        if K.op is not None:
+        if K.name == "qibotf":
```

Comment:
What do you think if we keep the `return func` in the qibotf backend class and remove this if condition?